### PR TITLE
fix(rollup): fixup dist and bundle in Buffer

### PR DESCRIPTION
**Pull request by edaniels**
Note: Undo bson dep once https://github.com/mongodb/js-bson/pull/260 is in

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,9 +54,11 @@
     "rollup": "^0.56.2",
     "rollup-plugin-babel": "^3.0.3",
     "rollup-plugin-commonjs": "^8.3.0",
+    "rollup-plugin-node-builtins": "^2.1.2",
     "rollup-plugin-node-resolve": "^3.0.3"
   },
   "dependencies": {
-    "bson": "^3.0.0"
+    "bson": "file:///Users/eric/js/js-bson",
```
Comment (edaniels) [inline package.json]: Will change after other PR goes in.

File: index.js [hunk 2]
```diff
@@ -1,3 +1,8 @@
 'use strict';
 const ExtJSON = require('./lib/ext_json');
-module.exports = ExtJSON;
+module.exports = {
```
Comment (edaniels) [inline index.js]: This disambiguates ESM output to be explicit about what the exports are.

File: rollup.config.js [hunk 3]
```diff
@@ -1,27 +1,97 @@
...
       ]
-    })
-  ],
-  external: ['bson'],
-}
+    ]
+  })
+];
+const external = Object.keys(pkg.dependencies || {});
+const defaultName = 'mongodb-extjson';
```
Comment (mbroadst) [inline rollup.config.js]: you mentioned this the other day, but I think we should standardize on calling this module `ejson`. Code examples should already be using `EJSON.parse`/`EJSON.stringify`, similar to the javascript `JSON` module. If you want to throw that into this PR I'd be fine with that

File: package.json (hunk 1 above)
Comment (mbroadst) [reply package.json]: :+1:

File: rollup.config.js (hunk 3 above)
Comment (edaniels) [reply rollup.config.js]: Discussed offline. Can only change this once we are okay with going to 4.x.

**Pull request merged**